### PR TITLE
Nearby Command: Obtain orders from specified region of Singapore (5 general regions)

### LINKED ISSUE #190: Modify nearby command to search for region
Modify nearby command to search for region: for example, north, south.

### PR DESCRIPTION
**Pull request by JeremyLoh**
This PR modifies/updates the following:
1. User Guide and Developer Guide for Nearby Command
2. Add ability for Nearby Command to search orders located at one of the 5 areas of Singapore
3. Update nearby command to use regex for searching orders
e.g. for postal sector 1 (S01), `.*S01\\d{4}.*`

For now, users can only search one of the 5 areas of Singapore: 
1. Central
2. East
3. North East
4. West
5. North

Information about the areas are obtained from:
https://keylocation.sg/singapore/districts-map

image

### CODE REVIEW

**Review by Exeexe93 — APPROVED** [review]
Looks good to merge!

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -365,20 +371,260 @@ https://www.ura.gov.sg/realEstateIIWeb/resources/misc/list_of_postal_districts.h
...
+====
+* You can view the orders located at a specified *area*.
+
+
+More information about the 5 areas of Singapore are shown below: +
+
+.Central
+|===
+|District |Areas |Location
+
```
Comment (khsc96) [inline docs/UserGuide.adoc]: I feel that this whole chunk can put in appendix and provide a link to it, at the same time at the appendix provide a link back to this particular section.

Something like
```
====
* You can view the orders located at a specified *area*.


More information about the 5 areas of Singapore are shown in <<appendix_region, Appendix region>>
```

...

```
[[appenix_region]]
the whole table here.
.
.
.
click <<nearby, here>> to go back to nearby command section
```
something like this?

File: src/main/java/seedu/address/logic/commands/NearbyCommand.java [hunk 2]
```diff
@@ -4,75 +4,153 @@
...
+ * Used to identify orders in the order book that belong to a given postal sector or area.
  */
 public class NearbyCommand extends Command {
     public static final String COMMAND_WORD = "nearby";
     public static final String NEWLINE = System.lineSeparator();
 
     public static final String MESSAGE_USAGE = COMMAND_WORD
             + ": View all orders located at the same postal sector based on the displayed order list."
-            + NEWLINE + "Parameters: POSTAL_SECTOR (must be a valid postal sector)"
+            + NEWLINE + "Parameters: POSTAL_SECTOR or Area"
```
Comment (khsc96) [inline src/main/java/seedu/address/logic/commands/NearbyCommand.java]: should this Area be AREA instead?

File: src/main/java/seedu/address/logic/commands/NearbyCommand.java [hunk 3]
```diff
@@ -4,75 +4,153 @@
...
+    public static final String MESSAGE_FAILURE_AREA = "Invalid area given.";
 
-    private final Index postalSector;
+    private final String searchTerm;
+    private final List<String> validAreas = new ArrayList<>(Arrays.asList(
+            "CENTRAL",
+            "EAST",
+            "NORTH-EAST",
+            "WEST",
+            "NORTH"));
```
Comment (khsc96) [inline src/main/java/seedu/address/logic/commands/NearbyCommand.java]: Is it better to make a enum class or perhaps like the CliSyntax of the original AB3 code? Is it possible anywhere else will be using region as well?

**Review by khsc96 — COMMENTED** [review]
Code looks very clean, thanks for refactoring some of the long chaining of command to cleaner code.
Suggested a few points, perhaps can take into considerations! Overall goodjob! 
If want to merge this version and nothing else to edit please let me know I'll integrate it into the main branch.

**Review by khsc96 — APPROVED** [review]
LGTM! good job!

**Pull request merged**